Add note regarding validation of nuget packages (#4882)
github/docs: content/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry.md

  dotnet restore
  ```

### Troubleshooting

> The NuGet packages *must* have their `RepositoryUrl` set to the current repository. The package feed will refuse the package if it is incorrect.

Reviewer: ```suggestion
Your NuGet package may fail to push if the `RepositoryUrl` in *.csproj* is not set to the expected repository .
```
